src/__tests__/PracticePage.test.tsx

```diff
@@ -27,11 +27,13 @@ describe('PracticePage', () => {
     fetchMock.resetMocks();
     fetchMock.mockResponseOnce(JSON.stringify(testDialogsResponse));
     mockSpeechRecognition = createMockSpeechRecognition();
-    wrapper = render(
-      <GlobalProvider speechRecognition={mockSpeechRecognition}/>
-        <PracticePage />
-      </GlobalProvider>
-    );
+    await act(async () => {
```

Comment:
This should definitely not be necessary. Any idea why it is?

Reply:
I assumed PracticePage makes a request? It doesn't matter for this suite, but it does for the RolePicker test. Rendering triggers a fetch request, that resolves after a microtask "tick", that then sets state. You need async act to capture that.

Reply:
Hmmm... How is that different from https://github.com/kentcdodds/react-act-debounced-counter/blob/master/src/__tests__/fetch-thing.js which is using the latest alpha and does not need async act?

Reply:
the async act that wraps findByText on the line following the render https://github.com/kentcdodds/react-act-debounced-counter/blob/master/src/__tests__/fetch-thing.js#L49 is what's catching the promise from the render.

Reply:
In that case, we should be covered here as well because the same thing applies to `waitForElementToBeRemoved` which is being used on the next line.

Reply:
>  It doesn't matter for this suite

and from the PR description 
> (I tried to fix practicepage.test.tsx too, but after fixing the syntax error, I couldn't figure out what props to pass component. Hope this helps as a starting point though.)

the failing test is in the other suite, not this one. Look at RolePicker.test.tsx

Reply:
I see. I think the correct solution here would be to add this before attempting anything else:

```javascript
await waitForElementToBeRemoved(() => getByText(/add roles to the dialog/i))
```

People really should be asserting on the loading state, and if they do that then they shouldn't need to worry about async act.